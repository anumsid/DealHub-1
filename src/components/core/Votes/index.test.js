import React from 'react'
import { shallow } from 'enzyme'
import { stub } from 'sinon'
import toJson from 'enzyme-to-json'
import { Votes } from './index'

describe('Votes Component', () => {
  let wrapper
  let onClick
  beforeEach(() => {
    onClick = stub()
    const props = {
      auth: 'xxx',
      onClick: onClick,
    }
    wrapper = shallow(<Votes {...props} />)
  })

  it('Matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('Dispatches click when clicked', () => {
    wrapper.find('Button').simulate('click')
    expect(onClick.calledOnce).toBe(true)
  })
})
