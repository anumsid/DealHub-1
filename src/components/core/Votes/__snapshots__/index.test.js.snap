// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Votes Component Matches snapshot 1`] = `
<div
  style={
    Object {
      "textAlign": "left",
    }
  }
>
  <Label
    color="pink"
    style={
      Object {
        "display": "inline-block",
      }
    }
  >
    <i
      aria-hidden={true}
      className="like icon"
    />
    0
  </Label>
  <Button
    as="button"
    color="pink"
    onClick={[Function]}
  >
    UpVote this Deal
  </Button>
</div>
`;
